src/arrayMethodJoin.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let separ = separator;
+    let str = '';
+
+    if (separ === undefined) {
+      separ = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (i === this.length - 1) {
+        str += this[i];
+      } else {
+        str += `${this[i]}${separ}`;
+      }
```

Comment:
add separator only when `i < this.length - 1`. 
```suggestion
      if (i < this.length - 1) {
        str += separator
      } 
```